boutdata/squashoutput.py

```diff
@@ -215,3 +258,180 @@ def squashoutput(
         # Note that get_chunk_cache() returns a tuple, so we have to unpack it when
         # passing to set_chunk_cache.
         set_chunk_cache(*netcdf4_chunk_cache)
+
+
+class SquashWorkers:
+    """
+    Class for packaging up worker processes for parallel writes, or passing the write
+    through in serial if parallel functionality not requested
+
+    Parameters
+    ----------
+    parallel : bool or int, default False
+        If False, write in serial. If True or 0 use all available processes. If positive
+        integer, use that many processes.
+    filenames : list of str
+        Names of the files to write to.
+    kwargs : dict
+        Keyword arguments to pass to DataFile constructors
+    """
+
+    def __init__(self, parallel, filenames, kwargs):
+        self.parallel = parallel
+        self.kwargs = kwargs
+        if self.parallel is False or len(filenames) == 1:
+            # No point doing parallel writing if there is only one output file
+            self.parallel = False
+            self.open_files(filenames)
+            return
+
+        if self.parallel is True or self.parallel == 0:
+            from boututils.run_wrapper import determineNumberOfCPUs
+
+            self.parallel = determineNumberOfCPUs()
+
+        self.n_outputs = len(filenames)
+
+        self.create_workers(filenames)
+
+    def __del__(self):
+        if self.parallel is False:
+            for f in self.files:
+                f.close()
+                del f
+        else:
+            for worker, connection, _ in self.workers:
+                # Send None to terminate worker process cleanly
+                connection.send(None)
+                worker.join()
+                connection.close()
+
+    def open_files(self, filenames):
+        """
+        Open files for serial writing
+
+        Parameters
+        ----------
+        filenames : list of str
+            Names of the files to write to.
+        """
+        from boututils.datafile import DataFile
```

Comment:
I see there are several imports in functions. I think PEP8 is recommending import at [the top of the file](https://www.python.org/dev/peps/pep-0008/#imports)

Reply:
```suggestion
        # use local imports to allow fast import for tab-completion
        from boututils.datafile import DataFile
```